sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -16,11 +16,115 @@
 
 package com.nvidia.spark.rapids
 
-import ai.rapids.cudf.{ColumnVector,GetJsonObjectOptions}
+import scala.util.parsing.combinator.RegexParsers
+
+import ai.rapids.cudf.{ColumnVector, GetJsonObjectOptions, Scalar}
 import com.nvidia.spark.rapids.Arm.withResource
 
-import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression}
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, GetJsonObject}
 import org.apache.spark.sql.types.{DataType, StringType}
+import org.apache.spark.unsafe.types.UTF8String
+
+// Copied from Apache Spark org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala
+private[this] sealed trait PathInstruction
+private[this] object PathInstruction {
+  case object Subscript extends PathInstruction
+  case object Wildcard extends PathInstruction
+  case object Key extends PathInstruction
+  case class Index(index: Long) extends PathInstruction
+  case class Named(name: String) extends PathInstruction
+}
+
+private[this] object JsonPathParser extends RegexParsers {
+  import PathInstruction._
+
+  def root: Parser[Char] = '$'
+
+  def long: Parser[Long] = "\\d+".r ^? {
+    case x => x.toLong
+  }
+
+  // parse `[*]` and `[123]` subscripts
+  def subscript: Parser[List[PathInstruction]] =
+    for {
+      operand <- '[' ~> ('*' ^^^ Wildcard | long ^^ Index) <~ ']'
+    } yield {
+      Subscript :: operand :: Nil
+    }
+
+  // parse `.name` or `['name']` child expressions
+  def named: Parser[List[PathInstruction]] =
+    for {
+      name <- '.' ~> "[^\\.\\[]+".r | "['" ~> "[^\\'\\?]+".r <~ "']"
+    } yield {
+      Key :: Named(name) :: Nil
+    }
+
+  // child wildcards: `..`, `.*` or `['*']`
+  def wildcard: Parser[List[PathInstruction]] =
+    (".*" | "['*']") ^^^ List(Wildcard)
+
+  def node: Parser[List[PathInstruction]] =
+    wildcard |
+      named |
+      subscript
+
+  val expression: Parser[List[PathInstruction]] = {
+    phrase(root ~> rep(node) ^^ (x => x.flatten))
+  }
+
+  def parse(str: String): Option[List[PathInstruction]] = {
+    this.parseAll(expression, str) match {
+      case Success(result, _) =>
+        Some(result)
+
+      case _ =>
+        None
+    }
+  }
+
+  def containsUnsupportedPath(instructions: List[PathInstruction]): Boolean = {
+    // Gpu GetJsonObject is not supported if JSON path contains wildcard [*]
+    // see https://github.com/NVIDIA/spark-rapids/issues/10216
+    instructions.exists {
+      case Wildcard => true
+      case Named(name) if name == "*" => true
```

Comment:
I don't think this is ever possible with how Spark currently parses the path, but it is better to be safe than sorry.

Reply:
Spark parse Wildcard with:
```
  // child wildcards: `..`, `.*` or `['*']`
  def wildcard: Parser[List[PathInstruction]] =
    (".*" | "['*']") ^^^ List(Wildcard)
```
So things like `. *` and `[ *]` are parsed to `Named(*)` and normalized to `['*']`. In this case, Spark and gpu are also mismatched in some cases:

`{"a":{"a":"d","*":"c"}}` and `$.a. *` => cpu: `c`, gpu: `["d","c"]`

Reply:
Thanks